### ISSUE TITLE
SVG savefig + LaTeX extremely slow on macOS

### ISSUE THREAD

**Issue by tdegeus** [matplotlib/matplotlib#9653]
When I saving a SVG figure is slowed down significantly (10-100x) when I use LaTeX, enabled by the following rc-setting:

```python
text.usetex : true
```

Note that rendering is somewhat slower, but not much. It is really the saving. I am on macOS 10.12.6 with matplotlib (2.1.0).

**Comment by dstansby** [comment]
Do you have a minimal example of code that when run causes the slowdown?

**Comment by tdegeus** [comment]
I have create two extremely simple examples, one without LaTeX and one with LaTeX (see below). The result:

```bash
$ time python3 without_latex.py
python3 without_latex.py  0.47s user 0.08s system 96% cpu 0.568 total
$ time python3 with_latex.py
python3 with_latex.py  6.40s user 0.62s system 96% cpu 7.251 total
```

**The use of LaTeX makes saving the figure 13x slower**.

The scripts:

*   `without_latex.py`

    ```python
    import matplotlib.pyplot as plt

    plt.figure()
    plt.plot([0,1],[0,1])
    plt.savefig('without_latex.svg')
    ```

*   `with_latex.py`

    ```python
    import matplotlib.pyplot as plt
    import matplotlib        as mpl

    mpl.rcParams['text.usetex'] = True

    plt.figure()
    plt.plot([0,1],[0,1])
    plt.savefig('with_latex.svg')
    ```

**Comment by tdegeus** [comment]
@jklymak Can you please verify the current description, and possibly lift the "needs clarification" label?

**Comment by jklymak** [comment]
I get about 3x slower.  

Not sure what the issue is.  It is a bit slower to `plt.show()` for me with latex, but not that much.  File sizes are the same, so its not like a font file is being dumped in there.

**Comment by tacaswell** [comment]
Using `latex` means we may be making (many) subprocess calls to latex which is my knee-jerk guess as to the source of the slow down.  We should be caching the results in `~/.cache/matplotlib/tex.cache` so repeated runs should get faster.  Are you seeing files accumulating there?

Do you see the slow down with other files types (png, pdf) or _just_ svg?

**Comment by jklymak** [comment]
With Master on a mac. 

The cache directory is not being written to on my machine.  

eps, png, pdf take about 2.8s.  
svg takes 7.0s

w/ usetex False they all take about 1s.  

I assume that the cache is not being written, and maybe svg is calling the subprocess twice for some reason.  ping @anntzer because I know he has/had a PR in to homogenize the backends.  Maybe that PR fixes the extra calls.  Not sure about the caching issue.

**Comment by jklymak** [comment]
OK. There appears to be a permissions issue on macOSX - the cache should be at `~/.matplotlib/tex.cache`.  But there is a test in `_get_config_or_cache_dir` that tests if `~/.matplotlib` is writable.  Which its not, I assume because someone decided to create the directory that way (not me).  So the cache was being written in `/var/tmp` in a new directory each time.

New timing is 5.2s for the second call to writing an svg.  1.8s for a pdf.

So the cache saves 2s for the svg, 1s for the PDF.

So, two problems:

  1. ~/.matplotlibrc directory should be created with the user writebit enabled (or the check in `_get_config_or_cache_dir` should be changed)
  2. SVG still seems quite slow and probably makes excess calls to TeX.

**Comment by jklymak** [comment]
A quick look at the backend code:  It looks like svg draws tex characters as paths.  It seems like the PDF gets the font info and places the characters.  Obviously drawing the paths will be a lot slower than just placing the characters.  Someone would have to spend some time on this I think.

**Comment by anntzer** [comment]
You can do `rcParams["svg.fonttype"] = "none"` to include text as text in SVG files.
Other than that, I don't think I have anything specific to help this case right now.

**Comment by tdegeus** [comment]
Possible duplicate: #4880

**Comment by tdegeus** [comment]
(sorry, clicked close instead of cancel)

**Comment by anntzer** [comment]
I think this will be closed by https://github.com/matplotlib/matplotlib/pull/19558 (or one of its variants), which appears to make svg+usetex essentially as fast as pdf+usetex (and much faster than either previous +usetex versions).

**Comment by anntzer** [comment]
Mostly closed by #19538 and #19558.  At least another possibility remaining for speeding up usetex is discussed at https://github.com/matplotlib/matplotlib/issues/4880#issuecomment-141684932, but that can be tracked by #4880.
